### PR TITLE
Add 1.4.0 to changelog

### PR DESCRIPTION
**Pull request by MatkovIvan**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,91 @@
+# 1.4.0 (April 2023)
+
+## Common
+
+- [Added support for Kotlin `1.8.20`](https://github.com/JetBrains/compose-multiplatform/pull/3000)
+- Update skiko from `0.7.54` to `0.7.58`
```
Comment (igordmn) [inline CHANGELOG.md:6]:
```suggestion
```
We usually don't mention skiko, just features from it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,91 @@
...
+- [Migrate skia to `m110`](https://github.com/JetBrains/skiko/pull/667)
```
Comment (igordmn) [inline CHANGELOG.md:7]:
```suggestion
```
Skia we don't mention as well

**Comment by m-sasha** [comment]
All the scrollbar fixes are missing:
- https://github.com/JetBrains/compose-multiplatform-core/pull/362
- https://github.com/JetBrains/compose-multiplatform-core/pull/365
- https://github.com/JetBrains/compose-multiplatform-core/pull/366
- https://github.com/JetBrains/compose-multiplatform-core/pull/372
- https://github.com/JetBrains/compose-multiplatform-core/pull/384
- https://github.com/JetBrains/compose-multiplatform-core/pull/387
- https://github.com/JetBrains/compose-multiplatform-core/pull/409

These can be summarized as

### Major Scrollbar overhaul
- Scrollbar now works correctly in lazy lists when items vary in size.
- Scrollbar now works correctly with lazy list contentPadding.
- Scrollbar now works correctly if the size of the scrollbar doesn't match the size of the widget it scrolls.
- Pressing the scrollbar track now works correctly.
- Added scrollbar support for lazy grid and text field.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,91 @@
...
+
+### Features
+
+- [Support perspective projection and `cameraDistance` parameter in `graphicsLayer`](https://github.com/JetBrains/compose-multiplatform-core/pull/422)
+- [Support `brush` parameter in `TextStyle` for complex coloring](https://github.com/JetBrains/compose-multiplatform/issues/2814)
+- [Support `drawStyle` parameter in `TextStyle` for drawing outlined text](https://github.com/JetBrains/compose-multiplatform-core/pull/470)
+- [Support `blendMode` parameter when drawing text on Canvas](https://github.com/JetBrains/compose-multiplatform-core/pull/470)
+- [Support compositing strategy on `graphicsLayer`](https://github.com/JetBrains/compose-multiplatform-core/pull/486)
+- [Add `minLines` parameter to `BasicTextField`](https://github.com/JetBrains/compose-multiplatform-core/pull/469)
+- [Support `painterResource` function common source set](https://github.com/JetBrains/compose-multiplatform/pull/2793)
```
Comment (igordmn) [inline CHANGELOG.md:17]:
```suggestion
- [Support `painterResource` function in common source set](https://github.com/JetBrains/compose-multiplatform/pull/2793)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,91 @@
...
+
+### Fixes
+
+- [Fix performing `ComposeScene.sendKeyEvent` an unintentional non-local return](https://github.com/JetBrains/compose-multiplatform-core/pull/398)
```
Comment (igordmn) [inline CHANGELOG.md:21]:
```suggestion
```
It is an internal fix

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,91 @@
...
+### Features
+
+- [Support external drag and drop](https://github.com/JetBrains/compose-multiplatform-core/pull/391)
+- [Animate scroll initiated by mouse wheel input](https://github.com/JetBrains/compose-multiplatform-core/pull/438)
+- [Remove AWT dependency in `rememberCursorPositionProvider`](https://github.com/JetBrains/compose-multiplatform-core/pull/432)
+
+### Fixes
+
+- [Fix showing a surface behind `AlertDialog` shape](https://github.com/JetBrains/compose-multiplatform/issues/1269)
+- [Fix `AlertDialog` shadow size](https://github.com/JetBrains/compose-multiplatform/issues/2857)
```
Comment (m-sasha) [inline CHANGELOG.md:39]: This fixes a bug introduced by another PR in the same version, so no need to mention it.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,91 @@
...
+- [Add `minLines` parameter to `BasicTextField`](https://github.com/JetBrains/compose-multiplatform-core/pull/469)
+- [Support `painterResource` function common source set](https://github.com/JetBrains/compose-multiplatform/pull/2793)
+
+### Fixes
+
+- [Fix performing `ComposeScene.sendKeyEvent` an unintentional non-local return](https://github.com/JetBrains/compose-multiplatform-core/pull/398)
+- [Actually remember provider in `rememberComponentRectPositionProvider`](https://github.com/JetBrains/compose-multiplatform-core/pull/444)
+- [Fix merging multiple selection areas inside `SelectionContainer`](https://github.com/JetBrains/compose-multiplatform/issues/2899)
+- [Fix crash on selecting text if layout result is invalid](https://github.com/JetBrains/compose-multiplatform-core/pull/482)
+- [Fix crash `SplitPane` with a modifier](https://github.com/JetBrains/compose-multiplatform/issues/2214)
```
Comment (igordmn) [inline CHANGELOG.md:25]: SplitPane is available only for desktop. Let's move it there

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,91 @@
...
+- [Fix `ContextMenuArea` to put the context menu representation in the same box with the content](https://github.com/JetBrains/compose-multiplatform/issues/2729)
+- [Replace using global mouse position with known position for context menus, dropdown menus and tooltips](https://github.com/JetBrains/compose-multiplatform-core/pull/437)
+- [Limit error dialog width in case of long error message](https://github.com/JetBrains/compose-multiplatform-core/pull/441)
+- [Fix window flashing background with unspecified size](https://github.com/JetBrains/compose-multiplatform-core/pull/442)
+- [Unregister window/dialog listeners when disposing](https://github.com/JetBrains/compose-multiplatform-core/pull/446)
+- [Fix applying `ComposePanel` visibility](https://github.com/JetBrains/compose-multiplatform/issues/2656)
+
+## Web
+
+- [`compose.web` is renamed to `compose.html`](https://github.com/JetBrains/compose-multiplatform/pull/2997)
```
Comment (igordmn) [inline CHANGELOG.md:53]:
```suggestion
- [`Compose for Web` is renamed to `Compose HTML Library`](https://github.com/JetBrains/compose-multiplatform/pull/2997)
```

**Comment by m-sasha** [comment]
[Fix window-sizing with `WindowState(size=Dp.Unspecified, undecorated=true, resizable=true)`](https://github.com/JetBrains/compose-multiplatform/issues/2676) is missing too.

File: CHANGELOG.md (hunk 7 above)
Comment (igordmn) [inline CHANGELOG.md]: Also, we should add:
- [The new experimental `Compose for Web` based on Kotlin/Wasm is available](README.md#web)

**Review by m-sasha — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (MatkovIvan) [reply CHANGELOG.md]: Already removed from previous feedback

**Comment by MatkovIvan** [comment]
> All the scrollbar fixes are missing

> https://github.com/JetBrains/compose-multiplatform/issues/2676 is missing too.

The mentioned fixes were included in `1.3.1`

**Review by igordmn — APPROVED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,95 @@
...
+
+### Features
+
+- [Support external drag and drop](https://github.com/JetBrains/compose-multiplatform-core/pull/391)
+- [Animate scroll initiated by mouse wheel input](https://github.com/JetBrains/compose-multiplatform-core/pull/438)
+- [Remove AWT dependency in `rememberCursorPositionProvider`](https://github.com/JetBrains/compose-multiplatform-core/pull/432)
+- Major scrollbar overhaul
+  - [Scrollbar now works correctly in lazy lists when items vary in size](https://github.com/JetBrains/compose-multiplatform/issues/2679)
+  - [Scrollbar now works correctly with lazy list `contentPadding`](https://github.com/JetBrains/compose-multiplatform/issues/2604)
+  - Scrollbar now works correctly in lazy list with spacing between items
```
Comment (m-sasha) [inline CHANGELOG.md:38]: https://github.com/JetBrains/compose-multiplatform-core/pull/380

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,95 @@
...
+  - Scrollbar now works correctly if the size of the scrollbar doesn't match the size of the widget it scrolls
```
Comment (m-sasha) [inline CHANGELOG.md:39]: https://github.com/JetBrains/compose-multiplatform-core/pull/368

**Review by m-sasha — APPROVED** [review]

File: CHANGELOG.md (hunk 9 above)
Comment (MatkovIvan) [reply CHANGELOG.md:39]:
```suggestion
  - [Scrollbar now works correctly if the size of the scrollbar doesn't match the size of the widget it scrolls](https://github.com/JetBrains/compose-multiplatform-core/pull/368)
```

**Pull request merged**